### DISCUSSION TITLE
extensions.ts in production got MODULE_NOT_FOUND

### DISCUSSION THREAD

**Discussion by MajedMuhammad** [directus/directus#15303] [Archive]
After building API, going to dist and run the application (simply by node start), getting this error (while previously was working fine): 

```
Error: Cannot find module '@directus/app/package.json'
Require stack:
- **\directusProject\packages\shared\dist\cjs\utils\node\resolve-package.js
- **\directusProject\packages\shared\dist\cjs\utils\node\get-extensions.js
- **\directusProject\packages\shared\dist\cjs\utils\node\index.js
- **\directusProject\api\dist\extensions.js
...
...
```

Probably this is caused due to it resolving the `@directus/app` not directly from `extensions` file but from an imported package's function `resolvePackage()` that exists in `packages\shared` which doesn't have `@directus/app` module as dependency. 

This idea appeared when I tried to resolve the current path inside the imported package's function `resolvePackage()` and the output looks like: 
```
**\directusProject\api\dist> node start 
Current path:  ****\directusProject\packages\shared\dist\cjs\utils\node\index.js
node:internal/modules/cjs/loader:933
  const err = new Error(message);
              ^

Error: Cannot find module '@directus/app/package.json'
...
...
```

Maybe after replacing npm by pnpm became importing 
```
import {
	ensureExtensionDirs,
	generateExtensionsEntry,
	getLocalExtensions,
	getPackageExtensions,
	resolvePackage,
} from '@directus/shared/utils/node';
```
not from dependencies `node_modules`, instead, looks like importing them directly from the workspace! 

Let me know your comments, I just dropped it here instead of creating a new issue because I have doubts about why nobody has reported this so far :). 
Thanks.

**Comment by azrikahar** [comment]
This seems similar to #14706, what version of Directus are you currently on and what OS/platform is it running on?

It was fixed by #14728, but there has been reports on it being an issue on Plesk, such as #15091.

**Reply by MajedMuhammad** [reply]
@azrikahar, thanks for replying. Your references are excellent and I got a clearer picture now PR #14728 fixed the issue. 
@rijkvanzanten, now the question replying to what is described in the PR #14728:
`Not really a bug, as 14706 was never officially supported`

What is the official way to run the API server in production? 
My assumption is to run the `dist` folder by `node/nodemon start.js`. Is there documentation to run it in diverse environments? 

Thanks a lot, Directus team.

**Reply by rijkvanzanten** [reply]
> What is the official way to run the API server in production?
> My assumption is to run the dist folder by node/nodemon start.js. Is there documentation to run it in diverse environments?

That assumption would be wrong 😁 The only two officially supported distributions are the final build on `npm`, and the Docker image on DockerHub 👍🏻 

You can run the npm one by running `npm install directus`, `directus init` (to setup the .env file, only required the first time), and then `directus start` to run it 👍🏻
